Kudu.Core/Scripts/selectNodeVersion.js

```diff
@@ -34,13 +34,26 @@ function resolveNpmPath(npmRootPath, npmVersion) {
 
 
 function getDefaultNpmVersion(nodeVersionPath) {
-    var npmLinkPath = path.resolve(nodeVersionPath, 'npm.txt');
-    // Determine if there's a link to npm at the node path
-    if (!existsSync(npmLinkPath)) {
-        return;
+    var appSettingNpmVersion = process.env.WEBSITE_NPM_DEFAULT_VERSION;
+
+    // extract the current node version from the path
+    var nodePathSplited = nodeVersionPath.split(path.sep).filter(function(e) { return e; });
+    var currentNodeVersion = nodePathSplited[nodePathSplited.length - 1];
+
+    if (appSettingNpmVersion) {
+        return appSettingNpmVersion;
+    } else if (currentNodeVersion === '4.1.2') {
```

Comment:
If they already map to a 3.x npm, then no. The problem with 4.1.2 is that it was the default node version for sites, but it maps to some 2.x version of NPM. And a vanilla ASP.NET 5 app was running into path length issue with that, so we force npm to be 3.3.6